manifests/client/install.pp

```diff
@@ -5,7 +5,19 @@
     fail 'class ::openldap::client has not been evaluated'
   }
 
-  package { $::openldap::client::package:
-    ensure => present,
+  case $::osfamily {
+    'Debian': {
+      package { $::openldap::client::package:
+        ensure => present,
+      }
+    }
+    'RedHat': {
+      class { '::openldap::client::utils':
```

Comment:
Using include will break the package parameter for openldap::client. This is why the unit tests fail in https://github.com/camptocamp/puppet-openldap/pull/114.

openldap::server already uses include, so there shouldn't be a duplicate declaration within the module. A duplicate declaration could occur if class is used outside the module (e.g. profiles, site.pp) to declare openldap::client and openldap::client::utils on a RedHat system. Avoiding that seems preferable to not being able to pass a custom the openldap::client::package in my opinion (especially since openldap::client exposes every current parameter for openldap::client::utils).